### ISSUE TITLE
FirebasePlugin does not conform to `FIRMessagingDelegate`

### ISSUE THREAD

**Issue by sebastianludwig** [arnesson/cordova-plugin-firebase#966]
The `FirebasePlugin` [sets itself](https://github.com/arnesson/cordova-plugin-firebase/blob/master/src/ios/FirebasePlugin.m#L117-L124) as `FIRMessaging` delegate. This [conflicts](https://github.com/arnesson/cordova-plugin-firebase/blob/master/src/ios/AppDelegate%2BFirebasePlugin.m#L70) with `AppDelegate+FirebasePlugin` actually [_being_](https://github.com/arnesson/cordova-plugin-firebase/blob/master/src/ios/AppDelegate%2BFirebasePlugin.m#L151-L157) the delegate. The assignment in `FirebasePlugin` even raises a compiler warning...

This prevents the plugin from properly handling data only push notifications. `AppDelegate+FirebasePlugin-messaging:didReceiveMessage:` is never called.

**Comment by konyavic** [comment]
We are facing a similar problem.
The root cause is in this plugin, it claims protocols of AppDelegate in a private extension.
The information of conforming any protocols in such way won't propagate to anywhere. (`conformsToProtocol:` returns NO)

Changing the private extension into a category should solve it without any harm:

```diff
-@interface AppDelegate () <UNUserNotificationCenterDelegate, FIRMessagingDelegate>
+@interface AppDelegate (FirebasePlugin) <UNUserNotificationCenterDelegate, FIRMessagingDelegate>
```

**Comment by sebastianludwig** [comment]
Good catch! However while correct I don't think this is enough to fix the issue: `FirebasePlugin` would still set itself as delegate and prevent the `AppDelegate` category from receiving any `FIRMessaging` delegate calls.

**Comment by konyavic** [comment]
Thank you for the correction!
